src/schema.ts

```diff
@@ -174,11 +174,14 @@ export class Schema {
       }
       // if the cardinality for the timeUnit is not cached, calculate it
       if (!fieldSchema.timeStats[encQ.timeUnit as string]) {
-        fieldSchema.timeStats[encQ.timeUnit as string] = timeSummary(encQ.timeUnit as TimeUnit, fieldSchema.stats);
+        fieldSchema.timeStats[encQ.timeUnit as string] = timeSummary(encQ.timeUnit as TimeUnit, fieldSchema.stats, excludeInvalid);
+      } else if (excludeInvalid) {
+        // we have it cached, but we have to make sure it excludes invalid values
+        return distinctExcluding(fieldSchema.timeStats[encQ.timeUnit as string], ['Invalid Date']);
```

Comment:
Why doesn't this include `null`?  Didn't we talk that `null` shouldn't even be converted in the date conversion method?